app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsSalableWithReservationsCondition.php

```diff
@@ -84,13 +84,22 @@ public function execute(string $sku, int $stockId, float $requestedQty): Product
             return $this->productSalableResultFactory->create(['errors' => $errors]);
         }
 
+        if(!(bool)$stockItemData['is_salable']){
+            $stockItemData['quantity'] = 0;
+        }
```

Comment:
it's better not to nullify the quantity of `$stockItemData[GetStockItemDataInterface::QUANTITY]` (ideally this array should be read-only) if the product is not salable but still proceed with computations 

```
        /** @var StockItemConfigurationInterface $stockItemConfiguration */
        $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);	        $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);
         $qtyWithReservation = $stockItemData[GetStockItemDataInterface::QUANTITY] +	        $qtyWithReservation = $stockItemData[GetStockItemDataInterface::QUANTITY] +
            $this->getReservationsQuantity->execute($sku, $stockId);	            $this->getReservationsQuantity->execute($sku, $stockId);
        $qtyLeftInStock = $qtyWithReservation - $stockItemConfiguration->getMinQty() - $requestedQty;	        $qtyLeftInStock = $qtyWithReservation - $stockItemConfiguration->getMinQty() - $requestedQty;
```

but instead to add conditional logic here:
```
        if ((bool)$stockItemData[GetStockItemDataInterface::IS_SALABLE]) {
            $qtyWithReservation = $stockItemData[GetStockItemDataInterface::QUANTITY] +
                $this->getReservationsQuantity->execute($sku, $stockId);
            $qtyLeftInStock = $qtyWithReservation - $stockItemConfiguration->getMinQty() - $requestedQty;
            $isEnoughQty = $qtyLeftInStock >= 0;
        } else {
            $isEnoughQty = false;
        }
```